Missed API reference

miguelangel-dev opened the pull request:
## Description
Complete API references for `nef`. It reaches the goal of 100% code coverage.

### Documented modules:
- `NefModels`: ProgressReport and CompilerOptions.

miguelangel-dev committed:
complete API reference

miguelangel-dev committed:
fix broken links

truizlop commented on project/Component/NefModels/CompilerOptions.swift:
```diff
@@ -2,13 +2,27 @@
 
+/// Swift compiler options
 public struct CompilerOptions {
+    /// List of the resources.
```
```suggestion
    /// Source files to compile.
```

truizlop commented on project/Component/NefModels/CompilerOptions.swift:
```diff
@@ -2,13 +2,27 @@
+    /// Initializes `CompilerOptions`
+    ///
+    /// - Parameters:
+    ///   - sources: list of the resources.
```
```suggestion
    ///   - sources: Source files to compile.
```

truizlop commented on project/Component/NefModels/CompilerOptions.swift:
```diff
@@ -2,13 +2,27 @@
+    ///
+    /// - Parameters:
+    ///   - sources: list of the resources.
+    ///   - platform: set SDK for the given platform.
```
```suggestion
    ///   - platform: Set SDK for the given platform.
```

truizlop commented on project/Component/NefModels/CompilerOptions.swift:
```diff
@@ -2,13 +2,27 @@
+    /// - Parameters:
+    ///   - sources: list of the resources.
+    ///   - platform: set SDK for the given platform.
+    ///   - frameworks: list of the framworks
```
```suggestion
    ///   - frameworks: List of the frameworks.
```

truizlop commented on project/Component/NefModels/CompilerOptions.swift:
```diff
@@ -2,13 +2,27 @@
+    ///   - sources: list of the resources.
+    ///   - platform: set SDK for the given platform.
+    ///   - frameworks: list of the framworks
+    ///   - linkers: list of the sources which should be passed to the linker.
```
```suggestion
    ///   - linkers: List of the sources which should be passed to the linker.
```

truizlop commented on project/Component/NefModels/CompilerOptions.swift:
```diff
@@ -2,13 +2,27 @@
+    ///   - platform: set SDK for the given platform.
+    ///   - frameworks: list of the framworks
+    ///   - linkers: list of the sources which should be passed to the linker.
+    ///   - libs: list of the libraries.
```
```suggestion
    ///   - libs: List of the libraries.
```

truizlop commented on project/Component/NefModels/ProgressReport.swift:
```diff
@@ -1,43 +1,67 @@
 import Bow
 import BowEffects
 
+/// Notify about the progress of an event.
```
```suggestion
/// Notifies the progress of an event.
```

truizlop commented on project/Component/NefModels/ProgressReport.swift:
```diff
@@ -1,43 +1,67 @@
 
+/// Notify about the progress of an event.
 public protocol ProgressReport {
+    /// Notify about the status of an event.
```
```suggestion
    /// Notifies the status of an event.
```

truizlop commented on project/Component/NefModels/ProgressReport.swift:
```diff
@@ -1,43 +1,67 @@
+/// Notify about the progress of an event.
 public protocol ProgressReport {
+    /// Notify about the status of an event.
+    /// - Parameter event: progress description.
```
```suggestion
    /// - Parameter event: Progress description.
```

truizlop commented on project/Component/NefModels/ProgressReport.swift:
```diff
@@ -1,43 +1,67 @@
 
 public extension ProgressReport {
+    /// Describes a succeeded event.
+    /// - Parameter step: step description.
```
```suggestion
    /// - Parameter step: Step description.
```

truizlop commented on project/Component/NefModels/ProgressReport.swift:
```diff
@@ -1,43 +1,67 @@
 }
 
 public extension ProgressReport {
+    /// Describes a succeeded event.
```
```suggestion
    /// Reports an event that completes successfully and instantly.
```

truizlop commented on project/Component/NefModels/ProgressReport.swift:
```diff
@@ -1,43 +1,67 @@
 public extension ProgressReport {
+    /// Describes a succeeded event.
+    /// - Parameter step: step description.
+    /// - Returns: An `IO` that represent the information for the succeeded event.
```
```suggestion
    /// - Returns: An `IO` describing the progress report.
```

truizlop commented on project/Component/NefModels/ProgressReport.swift:
```diff
@@ -1,43 +1,67 @@
 public protocol ProgressReport {
+    /// Notify about the status of an event.
+    /// - Parameter event: progress description.
+    /// - Returns: An `IO` that represent the information for the event.
```
```suggestion
    /// - Returns: An `IO` describing the progress report.
```

truizlop commented on project/Component/NefModels/ProgressReport.swift:
```diff
@@ -1,43 +1,67 @@
         self.inProgress(step).followedBy(self.succeeded(step))^
     }
     
+    /// Describes an event has just been initiated.
```
```suggestion
    /// Reports an ongoing event.
```

truizlop commented on project/Component/NefModels/ProgressReport.swift:
```diff
@@ -1,43 +1,67 @@
     }
     
+    /// Describes an event has just been initiated.
+    /// - Parameter step: step description.
```
```suggestion
    /// - Parameter step: Step description.
```

truizlop commented on project/Component/NefModels/ProgressReport.swift:
```diff
@@ -1,43 +1,67 @@
     
+    /// Describes an event has just been initiated.
+    /// - Parameter step: step description.
+    /// - Returns: An `IO` that represent the information for the initiated event.
```
```suggestion
    /// - Returns: An `IO` describing the progress report.
```

truizlop commented on project/Component/NefModels/ProgressReport.swift:
```diff
@@ -1,43 +1,67 @@
                           status: .inProgress))
     }
     
+    /// Describes the progress of an event finished with status `succeeded`.
```
```suggestion
    /// Reports the successful completion of an event.
```

truizlop commented on project/Component/NefModels/ProgressReport.swift:
```diff
@@ -1,43 +1,67 @@
     }
     
+    /// Describes the progress of an event finished with status `succeeded`.
+    /// - Parameter step: step description.
```
```suggestion
    /// - Parameter step: Step description.
```

truizlop commented on project/Component/NefModels/ProgressReport.swift:
```diff
@@ -1,43 +1,67 @@
     
+    /// Describes the progress of an event finished with status `succeeded`.
+    /// - Parameter step: step description.
+    /// - Returns: An `IO` that represent the information for the succeeded event.
```
```suggestion
    /// - Returns: An `IO` describing the progress report.
```

truizlop commented on project/Component/NefModels/ProgressReport.swift:
```diff
@@ -1,43 +1,67 @@
                           status: .successful))
     }
     
+    /// Describes the progress of an event finished with status `failed`.
```
```suggestion
    /// Reports the failed completion of an event.
```

truizlop commented on project/Component/NefModels/ProgressReport.swift:
```diff
@@ -1,43 +1,67 @@
     }
     
+    /// Describes the progress of an event finished with status `failed`.
+    /// - Parameter step: step description.
```
```suggestion
    /// - Parameter step: Step description.
```

truizlop commented on project/Component/NefModels/ProgressReport.swift:
```diff
@@ -1,43 +1,67 @@
     
+    /// Describes the progress of an event finished with status `failed`.
+    /// - Parameter step: step description.
+    /// - Returns: An `IO` that represent the information for the failed event.
```
```suggestion
    /// - Returns: An `IO` describing the progress report.
```

truizlop commented on project/Component/NefModels/ProgressReport.swift:
```diff
@@ -1,43 +1,67 @@
     }
 }
 
+/// Models the `progress status` for an event.
```
```suggestion
/// Models the progress status for an event.
```

truizlop commented on project/Component/NefModels/ProgressReport.swift:
```diff
@@ -1,43 +1,67 @@
     case failed
 }
 
+/// Models the `step` description for an event.
```
```suggestion
/// Describes the metadata associated with a progress event.
```

truizlop commented on project/Component/NefModels/ProgressReport.swift:
```diff
@@ -46,10 +70,18 @@ public extension CustomProgressDescription {
     var totalSteps: UInt { 1 }
 }
 
+/// Models an `event`.
```
```suggestion
/// Models an event that can be reported.
```

truizlop commented on project/Component/NefModels/ProgressReport.swift:
```diff
@@ -46,10 +70,18 @@ public extension CustomProgressDescription {
+    /// Describes the progress status.
     public let status: ProgressEventStatus
     
+    /// Initializes a `Platform`
```
```suggestion
    /// Initializes a `ProgressEvent`.
```

truizlop commented on project/Component/NefModels/ProgressReport.swift:
```diff
@@ -46,10 +70,18 @@ public extension CustomProgressDescription {
+    /// Initializes a `Platform`
+    ///
+    /// - Parameters:
+    ///   - step: current step.
```
```suggestion
    ///   - step: Current step.
```

truizlop commented on project/Component/NefModels/ProgressReport.swift:
```diff
@@ -46,10 +70,18 @@ public extension CustomProgressDescription {
+    ///
+    /// - Parameters:
+    ///   - step: current step.
+    ///   - status: progress status.
```
```suggestion
    ///   - status: Progress status.
```

miguelangel-dev committed:
Update project/Component/NefModels/CompilerOptions.swift

Co-authored-by: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CompilerOptions.swift

Co-authored-by: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/ProgressReport.swift

Co-authored-by: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/ProgressReport.swift

Co-authored-by: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/ProgressReport.swift

Co-authored-by: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/ProgressReport.swift

Co-authored-by: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/ProgressReport.swift

Co-authored-by: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CompilerOptions.swift

Co-authored-by: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CompilerOptions.swift

Co-authored-by: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/CompilerOptions.swift

Co-authored-by: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/ProgressReport.swift

Co-authored-by: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/ProgressReport.swift

Co-authored-by: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/ProgressReport.swift

Co-authored-by: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/ProgressReport.swift

Co-authored-by: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/ProgressReport.swift

Co-authored-by: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/ProgressReport.swift

Co-authored-by: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/ProgressReport.swift

Co-authored-by: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/ProgressReport.swift

Co-authored-by: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/ProgressReport.swift

Co-authored-by: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/ProgressReport.swift

Co-authored-by: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/ProgressReport.swift

Co-authored-by: Tomás Ruiz-López <truizlop@gmail.com>

Update project/Component/NefModels/ProgressReport.swift

Co-authored-by: Tomás Ruiz-López <truizlop@gmail.com>

capitalize parameters description in api docs

minor changes
